Release notes for 12672: Dynamic linking no-op on WASM (#1345)
bevyengine/bevy-website: release-content/0.14/release-notes/12672_Make_dynamic_linking_a_noop_on_wasm_targets.md

WASM does not support dynamic libraries that can be linked to during runtime. Before, Bevy would fail to compile if you enabled the `dynamic_linking` feature.

```bash
$ cargo build --target wasm32-unknown-unknown --features bevy/dynamic_linking
error: cannot produce dylib for `bevy_dylib v0.13.2` as the target `wasm32-unknown-unknown` does not support these crate types
```

> This has been changed so that Bevy will fallback to statically linking at compile time for all WASM platforms. If you enable `dynamic_linking` for development, you no longer need to disable it for WASM.

Reviewer: ```suggestion
Now, Bevy will fallback to statical linking for all WASM platforms. If you enable `dynamic_linking` for development, you no longer need to disable it for WASM.
```

even shorter :)
